scripts/dotnet-cli-build/PublishTargets.cs

```diff
@@ -436,5 +437,116 @@ public static BuildTargetResult TriggerDockerHubBuilds(BuildTargetContext c)
+
+            foreach (string file in AzurePublisherTool.ListBlobs(hostBlob))
+            {
+                var match = buildVersionRegex.Match(file);
+                if (match.Success)
+                {
+                    buildVersions.Add(match.Groups["version"].Value);
+                }
+            }
+
+            // Sort decending
+            buildVersions.Sort();
+            buildVersions.Reverse();
+
+            // Try to publish the last 10 builds
+            foreach (var bv in buildVersions.Take(10))
+            {
+                Console.WriteLine($"Checking drop version: {bv}");
+
+                if (ShouldDownloadAndPush(hostBlob, bv))
+                {
+                    DownloadPackagesForPush(hostBlob + bv);
+                }
+            }
+
+            return c.Success();
+        }
+
+        private static bool ShouldDownloadAndPush(string hostBlob, string buildVersion)
+        {
+            // Set of runtime ids to look for to act as the signal that the build
+            // as finished each of these legs of the build.
+            Dictionary<string, bool> runtimes = new Dictionary<string, bool>()
+            {
+                {"win7-x64", false },
+                {"win7-x86", false },
+                {"osx.10.10-x64", false },
+                {"rhel.7-x64", false },
+                {"ubuntu.14.04-x64", false },
+                {"debian.8-x64", false },
+            };
+
+            var buildFiles = AzurePublisherTool.ListBlobs(hostBlob + buildVersion);
+
+            foreach (var bf in buildFiles)
+            {
+                string buildFile = Path.GetFileName(bf);
+
+                if (buildFile == PackagePushedSemaphoreFileName)
+                {
+                    Console.WriteLine($"Found '{PackagePushedSemaphoreFileName}' for build version {buildVersion} so skipping this drop.");
+                    // Nothing to do because the latest build is uploaded.
+                    return false;
+                }
+
+                foreach (var runtime in runtimes.Keys)
+                {
+                    if (buildFile.StartsWith($"runtime.{runtime}"))
+                    {
+                        runtimes[runtime] = true;
+                        break;
+                    }
+                }
+            }
+
+            bool missingRuntime = false;
+            foreach (var runtime in runtimes)
+            {
+                if (!runtime.Value)
+                {
+                    missingRuntime = true;
+                    Console.WriteLine($"Version {buildVersion} missing packages for runtime {runtime.Key}");
+                }
+            }
+
+            if (missingRuntime)
+                Console.WriteLine($"Build version {buildVersion} is missing some runtime packages so not pushing this drop.");
+
+            return !missingRuntime;
+        }
+
+        private static void DownloadPackagesForPush(string pathToDownload)
```

Comment:
I realize the potential broken state may be unavoidable for the moment, I thought it was worth calling out though.